pkg/csi/service/cns/controller.go

```diff
@@ -252,6 +269,44 @@ func (c *controller) DeleteVolume(ctx context.Context, req *csi.DeleteVolumeRequ
 	if err != nil {
 		return nil, err
 	}
+	// Get the list of persistent volumes from k8s
+	pvList, err := c.pvLister.List(labels.Everything())
+	if err != nil {
+		msg := fmt.Sprintf("DeleteVolume: Error getting PersistentVolumes from API server with err: %v", err)
+		klog.Error(msg)
+		return nil, status.Errorf(codes.Internal, msg)
+	}
+	var pvFound bool
+	var pvName string
+	// Find the PV name for the delete volume request
+	for _, pv := range pvList {
+		if pv.Spec.CSI != nil && pv.Spec.CSI.VolumeHandle == req.VolumeId {
+			pvFound = true
+			pvName = pv.Name
+			klog.V(5).Infof("DeleteVolume: PersistentVolume %q found for the volume: %q", req.VolumeId)
+			break
+		}
+	}
+	if !pvFound {
+		msg := fmt.Sprintf("Unable to find the PersistentVolume from API server for the volume: %q:", req.VolumeId)
+		klog.Error(msg)
+		return nil, status.Errorf(codes.Internal, msg)
+	}
+	vaList, err := c.vaLister.List(labels.Everything())
```

Comment:
what is pupose of `label.Everything()`? Can we remove filtering?
if we want to add filtering we can filter based on PV name?

Reply:
c.vaLister.List() takes label selector as an argument. So, we need `labels.Everything()` in this case, since we do not depend on labels for VolumeAttachments
Reference:

```
// List lists all VolumeAttachments in the indexer.
List(selector labels.Selector) (ret []*v1.VolumeAttachment, err error)
```

Reply:
Also, verified that after restarting api server, the pvLister has the updated cache and delete volume request goes through successfully

```
 kubectl create -f pvc.yaml 
persistentvolumeclaim/example-vanilla-block-pvc created

root@k8s-master:~# kubectl get pvc -A
NAMESPACE   NAME                        STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS               AGE
default     example-vanilla-block-pvc   Bound    pvc-895df014-5e92-4b25-bf5a-1c118045fe9f   1Mi        RWO            example-vanilla-block-sc   7s

root@k8s-master:~# kubectl delete pod kube-apiserver-k8s-master -n kube-system
pod "kube-apiserver-k8s-master" deleted

root@k8s-master:~# kubectl delete -f pvc.yaml 

I1211 00:30:18.418326       1 controller.go:286] DeleteVolume: PersistentVolume "pvc-895df014-5e92-4b25-bf5a-1c118045fe9f" found for the volume: "0c0d7b12-1a12-440a-982d-5212a831154a"
```